test/integration/history_int_test.go

```diff
@@ -40,6 +44,54 @@ func (suite *HistoryIntegrationTestSuite) TestHistory_History() {
 	cp.ExpectExitCode(0)
 }
 
+func (suite *HistoryIntegrationTestSuite) TestRevert_failsOnCommitNotInHistory() {
+	suite.OnlyRunForTags(tagsuite.History)
+	ts := e2e.New(suite.T(), false)
+	defer ts.Close()
+
+	username := ts.CreateNewUser()
+
+	namespaceA := fmt.Sprintf("%s/%s", username, "revert-a")
+	wdA := createRevertProject(ts, namespaceA, true)
+
+	projFile, err := projectfile.FromPath(filepath.Join(wdA, "activestate.yaml"))
+	require.NoError(suite.T(), err, "cannot get current projectfile")
+
+	namespaceB := fmt.Sprintf("%s/%s", username, "revert-b")
+	wdB := createRevertProject(ts, namespaceB, false)
+
+	cp := ts.SpawnWithOpts(e2e.WithArgs("revert", projFile.CommitID()), e2e.WithWorkDirectory(wdB))
+	cp.SendLine("Y")
+	cp.Expect(projFile.CommitID())
+	cp.Expect("The target commit is not")
+	cp.ExpectNotExitCode(0)
+}
+
+func createRevertProject(ts *e2e.Session, namespace string, withUninstall bool) (workingDir string) {
```

Comment:
Just create the project under ActiveState-CI and reuse it for this test. We don't need to create it anew every time we run this test.

Reply:
I had considered that. If I understand correctly, using pre-configured projects would mean that any failure in this test will require manual intervention to ensure that the second project is put back into the correct state prior to the tests being run again. It also means that what that state needs to be is dependent on the first project. If, instead, we use one pre-configured and one dynamic, we can codify the state required, but it still feels a bit disjointed to me. It would also mean a little bit busier of a test (instead of using the same code to make two projects nearly the same, we'd need to checkout and use/shell the first project to get a commit that is useful to test with).

Reply:
I don't follow. `state revert` does not modify the project. What are you doing that's modifying the project?

My main concern with this is performance. The extra steps to create 2 projects will slow the tests down significantly.